app/assets/javascripts/stradivari/helpers.js

```diff
@@ -40,6 +40,27 @@ window._TABLE_ = (function() {
         concat(params)
       );
 
+    },
+
+    // example
+    // in: http://mysite.com?stradi_tabs[]=bar&stradi_tabs[]=baz&foo=bar
+    // out: {stradi_tabs: ['bar', 'baz'], foo: "bar"}
+    parseURLParameters: function(uri) {
+      var re = /([^&=]+)=?([^&]*)/g;
+      var decodeRE = /\+/g;  // Regex for replacing addition symbol with a space
+      var decode = function (str) {return decodeURIComponent( str.replace(decodeRE, " ") );};
+      var query = uri.split('?')[1]
```

Comment:
Managing URL parameters with reguar expresions is not  a great solution
from my point of view but it will work for now

On Mon, Jul 27, 2015 at 7:24 PM, Marcello Barnaba <notifications@github.com>
wrote:

> In app/assets/javascripts/stradivari/helpers.js
> <https://github.com/ifad/stradivari/pull/18#discussion_r35561990>:
>
> > @@ -40,6 +40,27 @@ window._TABLE_ = (function() {
> >          concat(params)
> >        );
> >
> > +    },
> > +
> > +    // example
> > +    // in: http://mysite.com?stradi_tabs[]=bar&stradi_tabs[]=baz&foo=bar
> > +    // out: {stradi_tabs: ['bar', 'baz'], foo: "bar"}
> > +    parseURLParameters: function(uri) {
> > +      var re = /([^&=]+)=?([^&]*)/g;
> > +      var decodeRE = /\+/g;  // Regex for replacing addition symbol with a space
> > +      var decode = function (str) {return decodeURIComponent( str.replace(decodeRE, " ") );};
> > +      var query = uri.split('?')[1]
>
> mixed feelings, do we really want to depend on the burden in taking that
> in the asset pipeline, or depending on a specific vendored version, or ...
>
> we have simple requirements, as such the simple code we have that it boils
> down to example.com?foo=bar&bar=baz".split('?')[1].split('&') should be
> enough... hairy regexen maybe not :-)
>
> wdyt?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ifad/stradivari/pull/18/files#r35561990>.
>



-- 
Lleïr Borràs i Metje
*http://about.me/lleir <http://about.me/lleir>*